cli/cmd/sudo/project/dump_resources.go

```diff
@@ -0,0 +1,161 @@
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			client, err := ch.Client()
+			if err != nil {
+				return err
+			}
+
+			var pattern string
+			// If args is not empty, use the first element as the pattern
+			if len(args) > 0 {
+				pattern = args[0]
+			} else {
+				pattern = "%"
+			}
+
+			res, err := client.SearchProjectNames(ctx, &adminv1.SearchProjectNamesRequest{
+				NamePattern: pattern,
+				Annotations: annotations,
+				PageSize:    pageSize,
+				PageToken:   pageToken,
+			})
+			if err != nil {
+				return err
+			}
+
+			if len(res.Names) == 0 {
+				ch.PrintfWarn("No projects found\n")
+				return nil
+			}
+
+			var m sync.Mutex
+			failedProjects := map[string]error{}
+			resources := map[string]map[string]json.RawMessage{}
+
+			grp, ctx := errgroup.WithContext(ctx)
+			for _, name := range res.Names {
+				org := strings.Split(name, "/")[0]
+				project := strings.Split(name, "/")[1]
+
+				grp.Go(func() error {
+					row, err := dumpResourcesForProject(ctx, client, org, project, typ)
+					if err != nil {
+						m.Lock()
+						failedProjects[name] = err
+						m.Unlock()
+						return nil
+					}
+					m.Lock()
+					resources[name] = row
+					m.Unlock()
+					return nil
+				})
+			}
+
+			err = grp.Wait()
+			if err != nil {
+				return err
+			}
+
+			// marshal as json with indentation
+			jsonData, err := json.MarshalIndent(resources, "", "  ")
+			if err != nil {
+				return fmt.Errorf("failed to marshal resources: %w", err)
+			}
+			ch.Println(string(jsonData))
+
+			for name, err := range failedProjects {
+				ch.Println()
+				ch.PrintfWarn("Failed to dump resources for project %v: %s\n", name, err)
+			}
+			if res.NextPageToken != "" {
+				ch.Println()
+				ch.Printf("Next page token: %s\n", res.NextPageToken)
+			}
+
+			return nil
+		},
+	}
+	searchCmd.Flags().StringVar(&typ, "type", "", "Filter for resources of a specific type")
+	searchCmd.Flags().StringToStringVar(&annotations, "annotation", nil, "Annotations to filter projects by (supports wildcard values)")
+	searchCmd.Flags().Uint32Var(&pageSize, "page-size", 1000, "Number of projects to return per page")
+	searchCmd.Flags().StringVar(&pageToken, "page-token", "", "Pagination token")
+
+	return searchCmd
+}
+
+func dumpResourcesForProject(ctx context.Context, c *client.Client, org, project, filter string) (map[string]json.RawMessage, error) {
+	proj, err := c.GetProject(ctx, &adminv1.GetProjectRequest{
+		OrganizationName:    org,
+		Name:                project,
+		IssueSuperuserToken: true,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	depl := proj.ProdDeployment
+	if depl == nil {
+		return nil, nil
+	}
+
+	rt, err := runtimeclient.New(depl.RuntimeHost, proj.Jwt)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to runtime: %w", err)
+	}
+
+	res, err := rt.ListResources(ctx, &runtimev1.ListResourcesRequest{InstanceId: depl.RuntimeInstanceId})
```

Comment:
If filtering by a type, you can pass it to the request here instead of checking locally